osu.Game/Overlays/Mods/ModPanel.cs

```diff
@@ -69,11 +72,54 @@ protected override void Deselect()
             Active.Value = false;
         }
 
+        /// <summary>
+        /// Determine if <see cref="ModPanel"/> is valid and can be shown
+        /// </summary>
+        public bool IsValid => modState.IsValid;
+
+        public bool ValidForSelection
+        {
+            get => modState.ValidForSelection.Value;
+            set
+            {
+                if (modState.ValidForSelection.Value == value)
+                    return;
+
+                modState.ValidForSelection.Value = value;
+            }
+        }
+
         #region Filtering support
 
+        public override IEnumerable<LocalisableString> FilterTerms => new[]
+        {
+            Mod.Name,
+            Mod.Acronym,
+            Mod.Description
+        };
+
+        public override bool MatchingFilter
+        {
+            get => modState.MatchingFilter.Value;
+            set
+            {
+                if (modState.MatchingFilter.Value == value)
+                    return;
+
+                modState.MatchingFilter.Value = value;
+            }
+        }
+
+        /// <summary>
+        /// This property is always <see langword="true"/> because it affects search result
+        /// </summary>
+        private readonly BindableBool canBeShown = new BindableBool(true);
+
+        IBindable<bool> IConditionalFilterable.CanBeShown => canBeShown;
```

Comment:
If this is going to be hardcoded to true, then why implement `IConditionalFilterable` at all?

Reply:
Changed to `IFilterable`